packages/lib/isOutOfBounds.tsx

```diff
@@ -286,23 +300,41 @@ export default function isOutOfBounds(
     bookerUtcOffset: number;
   },
   minimumBookingNotice?: number
-) {
-  return (
-    isTimeOutOfBounds({ time, minimumBookingNotice }) ||
-    isTimeViolatingFutureLimit({
-      time,
-      periodLimits: calculatePeriodLimits({
-        periodType,
-        periodDays,
-        periodCountCalendarDays,
-        periodStartDate,
-        periodEndDate,
-        // Temporary till we find a way to provide allDatesWithBookabilityStatus in handleNewBooking without re-computing availability for the booked timeslot
-        allDatesWithBookabilityStatusInBookerTz: null,
-        _skipRollingWindowCheck: true,
-        eventUtcOffset,
-        bookerUtcOffset,
-      }),
-    })
-  );
+): { isOutOfBounds: boolean; cause: string | null } {
+  const isOutOfBoundsByTime = isTimeOutOfBounds({ time, minimumBookingNotice });
+  const periodLimits = calculatePeriodLimits({
+    periodType,
+    periodDays,
+    periodCountCalendarDays,
+    periodStartDate,
+    periodEndDate,
+    allDatesWithBookabilityStatusInBookerTz: null, // Temporary workaround
+    _skipRollingWindowCheck: true,
+    eventUtcOffset,
+    bookerUtcOffset,
+  });
+
+  const isOutOfBoundsByPeriod = isTimeViolatingFutureLimit({
+    time,
+    periodLimits,
+  });
+
+  if (isOutOfBoundsByTime)
+    return {
+      isOutOfBounds: true,
+      cause: "Booking is out of bounds due to minimum booking notice.",
+    };
+
+  if (isOutOfBoundsByPeriod) {
+    log.warn(`Booking is out of bounds due to period restrictions - ${periodLimits}`);
+    return {
+      isOutOfBounds: true,
+      cause: "Booking is out of bounds due to period restrictions.",
+    };
+  }
+
+  return {
+    isOutOfBounds: false,
+    cause: null,
+  };
```

Comment:
Return cause along with isOutOfBounds to have a distinct reason on why a booking failed